app/app_xdg.go

```diff
@@ -32,6 +37,107 @@ func (a *fyneApp) OpenURL(url *url.URL) error {
 	return cmd.Start()
 }
 
+func findCurrentWM() string {
+
+	wm := os.Getenv("XDG_CURRENT_DESKTOP")
+	if wm == "" {
+		wm = os.Getenv("DESKTOP_SESSION")
+	}
+	wm = strings.ToLower(wm)
+	return wm
+}
+
+func findThemeVariant() fyne.ThemeVariant {
+	wm := findCurrentWM()
+	switch wm {
+	case "gnome", "xfce", "unity", "gnome-shell", "gnome-classic", "mate", "gnome-mate":
+		return findGnomeThemeVariant()
+	case "kde", "kde-plasma", "plasma":
+		return findKdeThemeVariant()
+	default:
+		return theme.VariantDark
+	}
+}
+
+// find the current KDE theme variant. At this time, no solution.
+func findKdeThemeVariant() fyne.ThemeVariant {
+	homedir, err := os.UserHomeDir()
+	if err != nil || homedir == "" {
+		// there is a problem, fallback to dark theme
+		return theme.VariantDark
+	}
+
+	// check if the user has a .config/kdeglobals file
+	kdeGlobals := filepath.Join(homedir, ".config/kdeglobals")
+	if _, err := os.Stat(kdeGlobals); os.IsNotExist(err) {
+		// no kdeglobals file, fallback to dark theme
+		return theme.VariantDark
+	}
+
+	// find the LookAndFeelPackage key in the kdeglobals file
+	content, err := ioutil.ReadFile(kdeGlobals)
+	if err != nil {
+		// there is a problem, fallback to dark theme
+		return theme.VariantDark
+	}
+	lines := strings.Split(string(content), "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, "LookAndFeelPackage=") {
+			// there is 2 possible values for the LookAndFeelPackage key
+			// =org.kde.breeze.desktop for the light theme
+			// =org.kde.breezedark.desktop for the dark theme
+			if strings.HasSuffix(line, "org.kde.breeze.desktop") {
+				return theme.VariantLight
+			} else if strings.HasSuffix(line, "org.kde.breezedark.desktop") {
+				return theme.VariantDark
+			}
+		}
+	}
+
+	// If we reach this point, it means that the LookAndFeelPackage key is not present in the kdeglobals file
+	// we can try to calculate the theme variant from the current KDE theme using the WM activeBackground key
+	for _, line := range lines {
+		if strings.HasPrefix(line, "activeBackground=") {
+			// read the value, it's a color in the form of r,g,b
+			col := strings.Split(line, "=")[1]
```

Comment:
As per your fyne-x bigger PR it would be good if this colour parsing was in a separate function.
We have the approach that a comment inline often shows that the code should have been in another function with appropriate name. Code is more resilient than docs ;)

Reply:
The newest code is pushed, is it OK for you?